### ISSUE TITLE
tests fail in 1-core VM

### ISSUE THREAD

**Issue by bmwiedemann** [libsemigroups/libsemigroups#251]
While working on [reproducible builds](https://reproducible-builds.org/) for [openSUSE](https://en.opensuse.org/openSUSE:Reproducible_Builds), I found that our `libsemigroups-1.3.3` package fails its tests in a 1-core VM.

Steps to reproduce on openSUSE or Debian:
```bash
osc co openSUSE:Factory/libsemigroups && cd $_
osc build --noservice --vm-type=kvm -j1 --clean standard
```

test output:
```
 Congruence 041: 2-sided congruence constructed from type only . . . 250μs
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 test_all is a Catch v2.9.2 host application.
 Run with -? for options 

 Congruence 042: no winner
 -----------------------------------------------------------------------
 tests/test-cong.cpp:1414
 .......................................................................

 tests/test-cong.cpp:1418: FAILED:
   REQUIRE( cong.word_to_class_index({2, 2, 2, 2}) == 254 )
 due to unexpected exception with message:
   src/todd-coxeter.cpp:757:run_impl: there are infinitely many classes in the
   congruence and Todd-Coxeter will never terminate

 Congruence 042: no winner . . .
 =======================================================================
 test cases:    156 |    155 passed | 1 failed
 assertions: 263514 | 263513 passed | 1 failed

**Comment by james-d-mitchell** [comment]
Thanks for the report @bmwiedemann, I have a pretty good idea what is causing this, and the fix should be straightforward, and I'll aim to make a release resolving this month.

**Comment by bmwiedemann** [comment]
Please ping me when you have a patch. I can test it in our env as well (unless cherry-picking to 1.3.3 is too hard)

**Comment by james-d-mitchell** [comment]
@bmwiedemann I believe I resolved this in v1.3.4, sorry I forgot to ping you earlier!

**Comment by bmwiedemann** [comment]
I verified that it is fixed in 1.3.4.
